controllers/account.controller.ts

```diff
@@ -0,0 +1,65 @@
+import { Request, Response } from "express";
+import { AccountService } from "../services/account.service";
+import { errorResponse, successResponse } from '../util/response.helper';
+import { ResponseMessage } from "../contants/response-message.contant";
+import { IAccount } from "../models/interface/account.model.interface";
+import { comparePassword, createToken } from "../util/helper";
+
+export class AccountController {
+
+    private accountService: AccountService;
+
+    constructor() {
+        this.accountService = new AccountService();
+    }
+
+    public async registerAccount(request: Request, response: Response) {
+        try {
+
+            const account: IAccount = await this.accountService.registerAccount(request.body);
+            return response.status(200).json(successResponse(ResponseMessage.User.successfullyRegistered, account));
+
+        } catch (error) {
+            console.error(`AccountController registerAccount error: ${error}`);
+            return response.status(500).json(errorResponse(ResponseMessage.User.notRegistered, error));
+        }
+    }
+
+    public async login(request: Request, response: Response) {
+        try {
+            const { email, password } = request.body;
+            const account = await this.accountService.findAccountByEmail(email);
+
+            if (!account) {
+                return response.status(404).json(errorResponse(ResponseMessage.User.emailNotFound));
+            }
+
+            const isMatch = await comparePassword(password, account.password);
+
+            if (!isMatch) {
+                return response.status(404).json(errorResponse(ResponseMessage.User.incorrectPassword));
+            }
+
+            const token = createToken(account);
+
+            const responseData = { account, token };
+
+            return response.status(201).json(successResponse(ResponseMessage.User.loginSuccessfull, responseData));
+
+        } catch (error) {
+            console.error(`AccountController login error: ${error}`);

```

Comment:
Please always try to include relevant details in the log in order to debug the issue, like you can improve this log like
`Error occurred in logging in user with username: ${username}, password: ${password}`. It's just an example. You can adjust it according to your need.

Reply:
logging enhancement will be a separate task